ably.d.ts

```diff
@@ -268,106 +543,249 @@ declare namespace Types {
+		/**
+		 * Maximum number of presence members to retrieve.
+		 */
 		limit?: number;
+		/**
+		 * When provided, will filter array of members returned that match the provided clientId string.
+		 */
 		clientId?: string;
+		/**
+		 * When provided, will filter array of members returned that match the provided connectionId string.
+		 */
 		connectionId?: string;
 	}
 
 	interface RealtimePresenceParams {
+		/**
+		 * When true (default) waits for the initial presence synchronization following channel attachment to complete before returning the members present. When false, the current list of members is returned without waiting for a complete synchronization.
+		 */
 		waitForSync?: boolean;
+		/**
+		 * When provided, will filter array of members returned that match the provided clientId string.
+		 */
 		clientId?: string;
+		/**
+		 * When provided, will filter array of members returned that match the provided connectionId string.
+		 */
 		connectionId?: string;
 	}
 
 	interface RealtimeHistoryParams {
+		/**
+		 * Earliest time in milliseconds since the epoch for any messages retrieved.
+		 */
 		start?: number;
+		/**
+		 * Latest time in milliseconds since the epoch for any messages retrieved.
+		 */
 		end?: number;
-		direction?: string;
+		/**
+		 * The direction to order messages retrieved. Defaults to backwards.
+		 */
+		direction?: 'forwards' | 'backwards';
+		/**
+		 * Maximum number of presence members to retrieve.
+		 */
 		limit?: number;
+		/**
+		 * When true, ensures message history is up until the point of the channel being attached. See [continuous history](https://ably.com/documentation/realtime/history#continuous-history) for more info. Requires the `direction` to be `backwards` (the default). If the `Channel` is not attached, or if `direction` is set to `forwards`, this option will result in an error.
+		 */
 		untilAttach?: boolean;
 	}
 
 	interface LogInfo {
 		/**
-		 * A number controlling the verbosity of the output. Valid values are: 0 (no logs), 1 (errors only),
-		 * 2 (errors plus connection and channel state changes), 3 (high-level debug output), and 4 (full debug output).
-		 **/
+		 * A number controlling the verbosity of the output. Valid values are: 0 (no logs), 1 (errors only), 2 (errors plus connection and channel state changes), 3 (high-level debug output), and 4 (full debug output).
+		 */
 		level?: number;
 
 		/**
-		 * A function to handle each line of log output. If handler is not specified, console.log is used.
-		 **/
+		 * A function to handle each line of log output. If handler is not specified, `console.log` is used.
+		 */
 		handler?: (msg: string) => void;
 	}
 
 	interface ChannelStateChange {
+		/**
+		 * The new current state.
+		 */
 		current: ChannelState;
+		/**
+		 * 	The previous state. (for the `update` event, this will be equal to the current state).
+		 */
 		previous: ChannelState;
+		/**
+		 * An ErrorInfo containing any information relating to the transition.
```

Comment:
```suggestion
		 * Information relating to the transition.
```

The type is implicit in the definition.